packages/cli-kit/src/public/node/api/admin.ts

```diff
@@ -0,0 +1,62 @@
+import {AdminSession} from '../../../session.js'
+import {content, token as outputToken} from '../../../output.js'
+import {Bug, Abort} from '../../../error.js'
+import {graphqlRequest} from '../../../private/common/api/graphql.js'
+import {ClientError, gql, RequestDocument, Variables} from 'graphql-request'
+
+export async function adminRequest<T>(
+  query: RequestDocument,
+  session: AdminSession,
+  variables?: Variables,
+): Promise<T> {
+  const api = 'Admin'
+  const version = await fetchApiVersion(session)
+  const url = adminUrl(session.storeFqdn, version)
+  return graphqlRequest(query, api, url, session.token, variables)
+}
+
+async function fetchApiVersion(session: AdminSession): Promise<string> {
```

Comment:
I'd recommend moving all the non-exported elements to `private/node/api/admin.ts`.

Reply:
this function is only used by the function above, I think it makes sense to have it here but not exported since it won't be used in any other places, what do you think?